helm/cluster-autoscaler-app-bundle/templates/iam.yaml

```diff
@@ -0,0 +1,58 @@
+{{- if eq .Values.provider "capa" }}
+apiVersion: iam.aws.upbound.io/v1beta1
+kind: Role
+metadata:
+  name: {{ .Values.clusterID }}-cluster-autoscaler
+  namespace: {{ .Release.Namespace }}
+  labels:
+    {{- include "cluster-autoscaler-app-bundle.labels" . | nindent 4 }}
+spec:
+  providerConfigRef:
+    name: {{ .Values.clusterID }}
+  forProvider:
+    tags:
+      giantswarm.io/cluster: {{ .Values.clusterID }}
+      sigs.k8s.io/cluster-api-provider-aws/cluster/{{ .Values.clusterID }}: owned
+      giantswarm.io/managed-by: {{ include "cluster-autoscaler-app-bundle.fullname" . }}
+    assumeRolePolicy: |
+      {
+        "Version": "2012-10-17",
+        "Statement": [
+          {{- include "cluster-autoscaler-app-bundle.trustPolicyStatements" . | nindent 10 }}
+        ]
+      }
+    inlinePolicy:
+      - name: cluster-autoscaler-policy
+        policy: |
+          {
+            "Version": "2012-10-17",
+            "Statement": [
```

Comment:
Let's remove the unnecessary duplicates that were in capa-iam-operator code. Only one block with `"Resource": "*"` is really needed.